springwolf-add-ons/springwolf-generic-binding/src/main/java/io/github/springwolf/addons/generic_binding/annotation/processor/PropertiesUtil.java

```diff
@@ -39,17 +43,33 @@ private static Map<String, Object> convertPropertiesToNestedMap(Properties prope
             Map<String, Object> mapNode = bindingData;
             while (path.size() > 1) {
                 String pathElement = path.get(0);
-                if (!mapNode.containsKey(pathElement)) {
-                    mapNode.put(pathElement, new HashMap<>());
-                }
-
+                mapNode.computeIfAbsent(pathElement, k -> new HashMap<>());
                 mapNode = (Map<String, Object>) mapNode.get(pathElement);
 
                 path.pop();
             }
 
-            mapNode.put(path.get(0), properties.get(propertyName));
+            var value = parseValue(properties.get(propertyName));
+            mapNode.put(path.get(0), value);
         }
         return bindingData;
     }
+
+    private static Object parseValue(Object input) {
+        if (input instanceof String inputString) {
+            Matcher matcher = ARRAY_PATTERN.matcher(inputString);
+
+            // Check if the pattern matches the input string
+            if (matcher.find()) {
+                // Extract the key and values
+                String[] values = matcher.group(1).split(",");
+                for (int i = 0; i < values.length; i++) {
+                    values[i] = values[i].trim();
+                }
+                return Arrays.asList(values);
```

Comment:
Nitpicking: can this 4 line block be written using Java stream api?

Reply:
Good catch, it's even nicer 😄  `return Arrays.stream(values).map(String::trim).toList();`